{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "374b8d39",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32f0e73e",
   "metadata": {},
   "outputs": [],
   "source": [
    "mitocarta = pd.read_csv(\"../data/public_db/mitocarta/Human.MitoCarta3.0_mito_genes.csv\", sep=\";\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "69dbc1d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HumanGeneID</th>\n",
       "      <th>MouseOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>liver_total_peak_intensity_log10</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1537</td>\n",
       "      <td>66445.0</td>\n",
       "      <td>CYC1</td>\n",
       "      <td>MC3DN6|UQCR4</td>\n",
       "      <td>cytochrome c1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>10,4</td>\n",
       "      <td>9,8</td>\n",
       "      <td>10,1</td>\n",
       "      <td>10,2</td>\n",
       "      <td>10,2</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>9,8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6390</td>\n",
       "      <td>67680.0</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>CWS2|IP|PGL4|SDH|SDH1|SDH2|SDHIP</td>\n",
       "      <td>succinate dehydrogenase complex iron sulfur su...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex II &gt; CII subunits | Metabolis...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10,2</td>\n",
       "      <td>10,3</td>\n",
       "      <td>9,2</td>\n",
       "      <td>10,1</td>\n",
       "      <td>10,3</td>\n",
       "      <td>9,9</td>\n",
       "      <td>9,9</td>\n",
       "      <td>10,1</td>\n",
       "      <td>9,9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10229</td>\n",
       "      <td>12850.0</td>\n",
       "      <td>COQ7</td>\n",
       "      <td>CAT5|CLK-1|CLK1|COQ10D8</td>\n",
       "      <td>coenzyme Q7, hydroxylase</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Metabolism &gt; Metals and cofactors &gt; Coenzyme Q...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>8,6</td>\n",
       "      <td>8,7</td>\n",
       "      <td>8,8</td>\n",
       "      <td>8,6</td>\n",
       "      <td>9</td>\n",
       "      <td>8,7</td>\n",
       "      <td>8,3</td>\n",
       "      <td>8,3</td>\n",
       "      <td>Plasma membrane (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6389</td>\n",
       "      <td>66945.0</td>\n",
       "      <td>SDHA</td>\n",
       "      <td>CMD1GG|FP|PGL5|SDH1|SDH2|SDHF</td>\n",
       "      <td>succinate dehydrogenase complex flavoprotein s...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex II &gt; CII subunits | Metabolis...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10,5</td>\n",
       "      <td>10,7</td>\n",
       "      <td>10,6</td>\n",
       "      <td>10,6</td>\n",
       "      <td>10,7</td>\n",
       "      <td>10,6</td>\n",
       "      <td>10,5</td>\n",
       "      <td>10,4</td>\n",
       "      <td>10,4</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7384</td>\n",
       "      <td>22273.0</td>\n",
       "      <td>UQCRC1</td>\n",
       "      <td>D3S3191|QCR1|UQCR1</td>\n",
       "      <td>ubiquinol-cytochrome c reductase core protein 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Protein import, sorting and homeostasis &gt; Prot...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>10,8</td>\n",
       "      <td>10,3</td>\n",
       "      <td>10,1</td>\n",
       "      <td>10,6</td>\n",
       "      <td>10,2</td>\n",
       "      <td>10,3</td>\n",
       "      <td>10,2</td>\n",
       "      <td>9,9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  HumanGeneID  MouseOrthologGeneID  Symbol                          Synonyms  \\\n",
       "0        1537              66445.0    CYC1                      MC3DN6|UQCR4   \n",
       "1        6390              67680.0    SDHB  CWS2|IP|PGL4|SDH|SDH1|SDH2|SDHIP   \n",
       "2       10229              12850.0    COQ7           CAT5|CLK-1|CLK1|COQ10D8   \n",
       "3        6389              66945.0    SDHA     CMD1GG|FP|PGL5|SDH1|SDH2|SDHF   \n",
       "4        7384              22273.0  UQCRC1                D3S3191|QCR1|UQCR1   \n",
       "\n",
       "                                         Description MitoCarta3.0_List  \\\n",
       "0                                      cytochrome c1      MitoCarta3.0   \n",
       "1  succinate dehydrogenase complex iron sulfur su...      MitoCarta3.0   \n",
       "2                           coenzyme Q7, hydroxylase      MitoCarta3.0   \n",
       "3  succinate dehydrogenase complex flavoprotein s...      MitoCarta3.0   \n",
       "4    ubiquinol-cytochrome c reductase core protein 1      MitoCarta3.0   \n",
       "\n",
       "                               MitoCarta3.0_Evidence  \\\n",
       "0  literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1  literature, APEX_matrix, targetP signal+, yeas...   \n",
       "2  literature, targetP signal+, yeast mito homolo...   \n",
       "3  literature, APEX_matrix, targetP signal+, yeas...   \n",
       "4  literature, APEX_matrix, targetP signal+, yeas...   \n",
       "\n",
       "  MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                              MIM   \n",
       "1                              MIM   \n",
       "2                              MIM   \n",
       "3                              MIM   \n",
       "4                              MIM   \n",
       "\n",
       "                           MitoCarta3.0_MitoPathways TrainingDataset  ...  \\\n",
       "0  OXPHOS > Complex III > CIII subunits | Metabol...           Tmito  ...   \n",
       "1  OXPHOS > Complex II > CII subunits | Metabolis...           Tmito  ...   \n",
       "2  Metabolism > Metals and cofactors > Coenzyme Q...           Tmito  ...   \n",
       "3  OXPHOS > Complex II > CII subunits | Metabolis...           Tmito  ...   \n",
       "4  Protein import, sorting and homeostasis > Prot...           Tmito  ...   \n",
       "\n",
       "  liver_total_peak_intensity_log10 heart_total_peak_intensity_log10  \\\n",
       "0                               10                             10,4   \n",
       "1                             10,2                             10,3   \n",
       "2                                9                              8,6   \n",
       "3                             10,5                             10,7   \n",
       "4                               10                             10,8   \n",
       "\n",
       "   skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                        9,8   \n",
       "1                                        9,2   \n",
       "2                                        8,7   \n",
       "3                                       10,6   \n",
       "4                                       10,3   \n",
       "\n",
       "  adipose_total_peak_intensity_log10  \\\n",
       "0                               10,1   \n",
       "1                               10,1   \n",
       "2                                8,8   \n",
       "3                               10,6   \n",
       "4                               10,1   \n",
       "\n",
       "  smallintestine_total_peak_intensity_log10  \\\n",
       "0                                      10,2   \n",
       "1                                      10,3   \n",
       "2                                       8,6   \n",
       "3                                      10,7   \n",
       "4                                      10,6   \n",
       "\n",
       "  largeintestine_total_peak_intensity_log10  \\\n",
       "0                                      10,2   \n",
       "1                                       9,9   \n",
       "2                                         9   \n",
       "3                                      10,6   \n",
       "4                                      10,2   \n",
       "\n",
       "   stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                  10                                   10   \n",
       "1                                 9,9                                 10,1   \n",
       "2                                 8,7                                  8,3   \n",
       "3                                10,5                                 10,4   \n",
       "4                                10,3                                 10,2   \n",
       "\n",
       "  testis_total_peak_intensity_log10  HPA_Main_Location_2020 (Reliability)  \n",
       "0                               9,8              Mitochondria (Supported)  \n",
       "1                               9,9              Mitochondria (Supported)  \n",
       "2                               8,3            Plasma membrane (Approved)  \n",
       "3                              10,4              Mitochondria (Supported)  \n",
       "4                               9,9              Mitochondria (Supported)  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mitocarta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1582fe85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['HumanGeneID', 'MouseOrthologGeneID', 'Symbol', 'Synonyms',\n",
      "       'Description', 'MitoCarta3.0_List', 'MitoCarta3.0_Evidence',\n",
      "       'MitoCarta3.0_SubMitoLocalization', 'MitoCarta3.0_MitoPathways',\n",
      "       'TrainingDataset', 'MitoCarta2.0_Score', 'MitoCarta2.0_FDR',\n",
      "       'MitoCarta2.0_List', 'MitoCarta2.0_Evidence',\n",
      "       'EnsemblGeneID_mapping_version_20200130', 'UniProt', 'ProteinLength',\n",
      "       'TargetP_Score', 'MitoDomain_Score', 'CoexpressionGnfN50_Score',\n",
      "       'PGC_Induction_Score', 'YeastMitoHomolog_Score',\n",
      "       'RickettsiaHomolog_Score', 'MSMS_Score', 'hg19_Chromosome',\n",
      "       'hg19_Start', 'hg19_Stop', 'MSMS_NUM_TISSUES',\n",
      "       'MSMS_NUM_PEPTIDES_UNIQUE', 'MSMS_NUM_SPECTRA', 'MSMS_TOTAL_INTENSITY',\n",
      "       'MSMS_PERCENT_COVERAGE', 'Tissues',\n",
      "       'cerebrum_total_peak_intensity_log10',\n",
      "       'cerebellum_total_peak_intensity_log10',\n",
      "       'brainstem_total_peak_intensity_log10',\n",
      "       'spinalcord_total_peak_intensity_log10',\n",
      "       'kidney_total_peak_intensity_log10', 'liver_total_peak_intensity_log10',\n",
      "       'heart_total_peak_intensity_log10',\n",
      "       'skeletalmuscle_total_peak_intensity_log10',\n",
      "       'adipose_total_peak_intensity_log10',\n",
      "       'smallintestine_total_peak_intensity_log10',\n",
      "       'largeintestine_total_peak_intensity_log10',\n",
      "       'stomach_total_peak_intensity_log10',\n",
      "       'placenta_total_peak_intensity_log10',\n",
      "       'testis_total_peak_intensity_log10',\n",
      "       'HPA_Main_Location_2020 (Reliability)'],\n",
      "      dtype='object')\n",
      "(1157, 48)\n"
     ]
    }
   ],
   "source": [
    "print(mitocarta.columns)\n",
    "print(mitocarta.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "95a33285",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1136 genes uniques\n",
      "1136 symbols uniques\n"
     ]
    }
   ],
   "source": [
    "print(mitocarta[\"HumanGeneID\"].nunique(), \"genes uniques\")\n",
    "print(mitocarta[\"Symbol\"].nunique(), \"symbols uniques\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c5053228",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1136\n"
     ]
    }
   ],
   "source": [
    "print(mitocarta[\"EnsemblGeneID_mapping_version_20200130\"].nunique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99715637",
   "metadata": {},
   "source": [
    "### Fonctions utilitaires"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd643a13",
   "metadata": {},
   "outputs": [],
   "source": [
    "import unicodedata\n",
    "\n",
    "def normalize_text(s):\n",
    "    \"\"\"Supprime espaces et accents, normalise le texte\"\"\"\n",
    "    if pd.isna(s):\n",
    "        return None\n",
    "    s = str(s).strip()\n",
    "    s = unicodedata.normalize('NFKD', s)\n",
    "    return s\n",
    "\n",
    "import re\n",
    "\n",
    "def split_field_to_list(s):\n",
    "    \"\"\"Transforme un champ sÃ©parÃ© par ; , | / en liste\"\"\"\n",
    "    if pd.isna(s):\n",
    "        return []\n",
    "    parts = re.split(r'[;,|/]+', str(s))\n",
    "    return [p.strip() for p in parts if p.strip()]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "48d5fbbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0                             [MC3DN6, UQCR4]\n",
      "1    [CWS2, IP, PGL4, SDH, SDH1, SDH2, SDHIP]\n",
      "2                [CAT5, CLK-1, CLK1, COQ10D8]\n",
      "3        [CMD1GG, FP, PGL5, SDH1, SDH2, SDHF]\n",
      "4                      [D3S3191, QCR1, UQCR1]\n",
      "Name: Synonyms_list, dtype: object\n"
     ]
    }
   ],
   "source": [
    "# DÃ©tection de la structure et du nettoyage Ã  appliquer.\n",
    "mitocarta['Synonyms_list'] = mitocarta['Synonyms'].apply(split_field_to_list)\n",
    "mitocarta['MitoCarta3_Pathways_list'] = mitocarta['MitoCarta3.0_MitoPathways'].apply(split_field_to_list)\n",
    "mitocarta['MitoCarta3_SubMitoLoc_list'] = mitocarta['MitoCarta3.0_SubMitoLocalization'].apply(split_field_to_list)\n",
    "\n",
    "print(mitocarta[\"Synonyms_list\"].head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "04e072d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "pivot_records = {}\n",
    "\n",
    "for _, row in mitocarta.iterrows():\n",
    "    hgnc_id = row['HumanGeneID']\n",
    "    if pd.isna(hgnc_id):\n",
    "        continue  # sauter les lignes sans HumanGeneID\n",
    "\n",
    "    record = {\n",
    "        \"_key\": hgnc_id,\n",
    "        \"human_gene_id\": hgnc_id,\n",
    "        \"symbol\": normalize_text(row['Symbol']),\n",
    "        \"synonyms\": split_field_to_list(row['Synonyms']),\n",
    "        \"description\": normalize_text(row['Description']),\n",
    "        \"ensembl_id\": normalize_text(row['EnsemblGeneID_mapping_version_20200130']),\n",
    "        \"uniprot_ids\": split_field_to_list(row['UniProt']),\n",
    "        \"mouse_ortholog_gene_id\": normalize_text(row['MouseOrthologGeneID']),\n",
    "        \n",
    "        \"mitocarta3\": {\n",
    "            \"list_status\": row['MitoCarta3.0_List'] == \"yes\",\n",
    "            \"evidence\": split_field_to_list(row['MitoCarta3.0_Evidence']),\n",
    "            \"sub_mito_localization\": split_field_to_list(row['MitoCarta3.0_SubMitoLocalization']),\n",
    "            \"mito_pathways\": split_field_to_list(row['MitoCarta3.0_MitoPathways'])\n",
    "        },\n",
    "\n",
    "        \"mitocarta2\": {\n",
    "            \"score\": row['MitoCarta2.0_Score'],\n",
    "            \"fdr\": row['MitoCarta2.0_FDR'],\n",
    "            \"list_status\": row['MitoCarta2.0_List'] == \"yes\",\n",
    "            \"evidence\": split_field_to_list(row['MitoCarta2.0_Evidence'])\n",
    "        },\n",
    "\n",
    "        \"training_dataset\": row['TrainingDataset'],\n",
    "        \"protein_length\": row['ProteinLength'],\n",
    "        \"scores\": {\n",
    "            \"targetp\": row['TargetP_Score'],\n",
    "            \"mitodomain\": row['MitoDomain_Score'],\n",
    "            \"coexpression\": row['CoexpressionGnfN50_Score'],\n",
    "            \"pgc_induction\": row['PGC_Induction_Score'],\n",
    "            \"yeast_homolog\": row['YeastMitoHomolog_Score'],\n",
    "            \"rickettsia_homolog\": row['RickettsiaHomolog_Score']\n",
    "        },\n",
    "\n",
    "        \"msms\": {\n",
    "            \"score\": row['MSMS_Score'],\n",
    "            \"num_tissues\": row['MSMS_NUM_TISSUES'],\n",
    "            \"num_peptides_unique\": row['MSMS_NUM_PEPTIDES_UNIQUE'],\n",
    "            \"num_spectra\": row['MSMS_NUM_SPECTRA'],\n",
    "            \"total_intensity\": row['MSMS_TOTAL_INTENSITY'],\n",
    "            \"percent_coverage\": row['MSMS_PERCENT_COVERAGE'],\n",
    "            \"by_tissue_log10\": {\n",
    "                \"cerebrum\": row['cerebrum_total_peak_intensity_log10'],\n",
    "                \"cerebellum\": row['cerebellum_total_peak_intensity_log10'],\n",
    "                \"brainstem\": row['brainstem_total_peak_intensity_log10'],\n",
    "                \"spinalcord\": row['spinalcord_total_peak_intensity_log10'],\n",
    "                \"kidney\": row['kidney_total_peak_intensity_log10'],\n",
    "                \"liver\": row['liver_total_peak_intensity_log10'],\n",
    "                \"heart\": row['heart_total_peak_intensity_log10'],\n",
    "                \"skeletalmuscle\": row['skeletalmuscle_total_peak_intensity_log10'],\n",
    "                \"adipose\": row['adipose_total_peak_intensity_log10'],\n",
    "                \"smallintestine\": row['smallintestine_total_peak_intensity_log10'],\n",
    "                \"largeintestine\": row['largeintestine_total_peak_intensity_log10'],\n",
    "                \"stomach\": row['stomach_total_peak_intensity_log10'],\n",
    "                \"placenta\": row['placenta_total_peak_intensity_log10'],\n",
    "                \"testis\": row['testis_total_peak_intensity_log10']\n",
    "            }\n",
    "        },\n",
    "\n",
    "        \"genome_coords_hg19\": {\n",
    "            \"chromosome\": row['hg19_Chromosome'],\n",
    "            \"start\": row['hg19_Start'],\n",
    "            \"stop\": row['hg19_Stop']\n",
    "        },\n",
    "\n",
    "        \"hpa_location_2020\": row['HPA_Main_Location_2020 (Reliability)']\n",
    "    }\n",
    "\n",
    "    pivot_records[hgnc_id] = record\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a095d8a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/6n/hw7c94m14pn9syslmn86py080000gq/T/ipykernel_86291/2318277566.py:10: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).\n",
      "  rec['_created_at'] = datetime.datetime.utcnow().isoformat() + 'Z'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pivot JSONL sauvegardÃ© : ./pivot_output/mitocarta_pivot.jsonl\n",
      "Pivot JSON complet sauvegardÃ© : ./pivot_output/mitocarta_pivot_full.json\n"
     ]
    }
   ],
   "source": [
    "import json, os, datetime\n",
    "\n",
    "outdir = \"../data/pivot_output\"\n",
    "os.makedirs(outdir, exist_ok=True)\n",
    "\n",
    "# JSONL (une ligne par gene)\n",
    "jsonl_path = os.path.join(outdir, \"mitocarta_pivot.jsonl\")\n",
    "with open(jsonl_path, 'w', encoding='utf8') as f:\n",
    "    for rec in pivot_records.values():\n",
    "        rec['_created_at'] = datetime.datetime.utcnow().isoformat() + 'Z'\n",
    "        f.write(json.dumps(rec, ensure_ascii=False) + \"\\n\")\n",
    "\n",
    "print(\"Pivot JSONL sauvegardÃ© :\", jsonl_path)\n",
    "\n",
    "# JSON complet\n",
    "json_path = os.path.join(outdir, \"mitocarta_pivot_full.json\")\n",
    "with open(json_path, 'w', encoding='utf8') as f:\n",
    "    json.dump(pivot_records, f, ensure_ascii=False, indent=2)\n",
    "\n",
    "print(\"Pivot JSON complet sauvegardÃ© :\", json_path)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "cluster_analysis",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
